### PR TITLE
Remove legacy metrics client from podautoscaler

### PR DESCRIPTION
**Pull request by serathius**
/kind cleanup

```release-note
kube-controller-manger: the `--horizontal-pod-autoscaler-use-rest-clients`  flag and Heapster support in the horizontal pod autoscaler, deprecated since 1.12, is removed.
```
/sig instrumentation
/sig autoscaling
/priority important-longterm
/assign @MaciekPytel

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/hpacontroller.go [hunk 1]
```diff
@@ -40,9 +40,7 @@ func (o *HPAControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&o.HorizontalPodAutoscalerDownscaleForbiddenWindow.Duration, "horizontal-pod-autoscaler-downscale-delay", o.HorizontalPodAutoscalerDownscaleForbiddenWindow.Duration, "The period since last downscale, before another downscale can be performed in horizontal pod autoscaler.")
 	fs.MarkDeprecated("horizontal-pod-autoscaler-downscale-delay", "This flag is currently no-op and will be deleted.")
 	fs.Float64Var(&o.HorizontalPodAutoscalerTolerance, "horizontal-pod-autoscaler-tolerance", o.HorizontalPodAutoscalerTolerance, "The minimum change (from 1.0) in the desired-to-actual metrics ratio for the horizontal pod autoscaler to consider scaling.")
-	fs.BoolVar(&o.HorizontalPodAutoscalerUseRESTClients, "horizontal-pod-autoscaler-use-rest-clients", o.HorizontalPodAutoscalerUseRESTClients, "If set to true, causes the horizontal pod autoscaler controller to use REST clients through the kube-aggregator, instead of using the legacy metrics client through the API server proxy.  This is required for custom metrics support in the horizontal pod autoscaler.")
```
Comment (logicalhan) [inline cmd/kube-controller-manager/app/options/hpacontroller.go:43]: Is this safe to remove? How long has this flag been deprecated? (https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli)

**Comment by logicalhan** [comment]
I think you need to run `hack/update-bazel.sh` and possibly `hack/update-vendor.sh`

File: cmd/kube-controller-manager/app/options/hpacontroller.go (hunk 1 above)
Comment (serathius) [reply cmd/kube-controller-manager/app/options/hpacontroller.go:43]: https://github.com/kubernetes/kubernetes/pull/68089
1.12

**Comment by serathius** [comment]
ping @lavalamp
/cc @liggitt

**Comment by liggitt** [comment]
drop the lines in `cluster/gce/gci/configure-helper.sh` that set `--horizontal-pod-autoscaler-use-rest-clients`

lgtm otherwise

**Comment by serathius** [comment]
Done

**Pull request merged**